tests/rkt_api_service_test.go

```diff
@@ -0,0 +1,297 @@
+			return
+		}
+		t.Fatalf("Unexpected state: %v", rawState)
+	}
+	t.Errorf("Pod state returned by api-service (%q) is not equivalent to the state returned by 'rkt status' (%q)", apiState, rawState)
+}
+
+func checkPodApps(t *testing.T, rawApps []*appInfo, apiApps []*v1alpha.App, checkAppState bool) {
+	if len(rawApps) != len(apiApps) {
+		t.Errorf("Expected %d apps, saw %d apps returned by api service %v", len(rawApps), len(apiApps), apiApps)
+	}
+
+	for _, m := range rawApps {
+		found := false
+		for _, n := range apiApps {
+			if m.name == n.Name {
+				found = true
+				if checkAppState && m.exitCode != int(n.ExitCode) {
+					t.Errorf("Expected %v, saw %v", m.exitCode, n.ExitCode)
+				}
+				if m.image.id != n.Image.Id {
+					t.Errorf("Expected %q, saw %q", m.image.id, n.Image.Id)
+				}
+				break
+			}
+		}
+		if !found {
+			t.Errorf("Expected app (name: %q) in the app list", m.name)
+		}
+	}
+}
+
+func checkPodNetworks(t *testing.T, rawNets []*networkInfo, apiNets []*v1alpha.Network) {
+	if len(rawNets) != len(apiNets) {
+		t.Errorf("Expected %d networks, saw %d networks returned by api service", len(rawNets), len(apiNets))
+	}
+
+	// Each network should have a unique name, so iteration over one list is enough given
+	// the lengths of the two lists are equal.
+	for _, m := range rawNets {
+		found := false
+		for _, n := range apiNets {
+			if m.name == n.Name {
+				found = true
+				if m.ipv4 != n.Ipv4 {
+					t.Errorf("Expected %q, saw %q", m.ipv4, n.Ipv4)
+				}
+				break
+			}
+		}
+		if !found {
+			t.Errorf("Expected network (name: %q, ipv4: %q) in networks", m.name, m.ipv4)
+		}
+	}
+}
+
+// Check the pod's information by 'rkt status'.
+func checkPod(t *testing.T, ctx *rktRunCtx, p *v1alpha.Pod, checkAppState bool) {
+	podInfo := getPodInfo(t, ctx, p.Id)
+	if podInfo.id != p.Id {
+		t.Errorf("Expected %q, saw %q", podInfo.id, p.Id)
+	}
+	if podInfo.pid != int(p.Pid) {
+		t.Errorf("Expected %q, saw %q", podInfo.pid, p.Pid)
+	}
+	checkPodState(t, podInfo.state, p.State)
+	checkPodApps(t, podInfo.apps, p.Apps, checkAppState)
+	checkPodNetworks(t, podInfo.networks, p.Networks)
+
+	// TODO(yifan): Check manifest only when testing InspectPod().
+	if p.Manifest != nil && !bytes.Equal(podInfo.manifest, p.Manifest) {
+		t.Errorf("Expected %q, saw %q", string(podInfo.manifest), string(p.Manifest))
+	}
+}
+
+// Check the image's information by 'rkt image list'.
+func checkImage(t *testing.T, ctx *rktRunCtx, m *v1alpha.Image) {
+	imgInfo := getImageInfo(t, ctx, m.Id)
+	if imgInfo.id != m.Id {
+		t.Errorf("Expected %q, saw %q", imgInfo.id, m.Id)
+	}
+	if imgInfo.name != m.Name {
+		t.Errorf("Expected %q, saw %q", imgInfo.name, m.Name)
+	}
+	if imgInfo.version != m.Version {
+		t.Errorf("Expected %q, saw %q", imgInfo.version, m.Version)
+	}
+	if imgInfo.importTime != m.ImportTimestamp {
+		t.Errorf("Expected %q, saw %q", imgInfo.importTime, m.ImportTimestamp)
+	}
+
+	// TODO(yifan): Check manifest only when testing InspectImage().
```

Comment:
@krnowak Yeah, I was kind waiting #1688 . If either one gets merged first, I will change this to take a bool to determine if we need to check the manifest in the function.